### PR TITLE
Set values for datatypes integer, boolean, string in continuous_time_…

### PR DESCRIPTION
**Pull request by nik5346**
#10915

### CODE REVIEW

**Review by AnHeuermann — CHANGES_REQUESTED** [review]
I don't think violating the FMI standard (2.0.4) is the correct solution for the problem you described in #10915.

File: OMCompiler/SimulationRuntime/c/fmi/FMI2Common.c [hunk 1]
```diff
@@ -156,7 +156,7 @@ void fmi2SetInteger_OMC(void* in_fmi2, int numberOfValueReferences, double* inte
 {
   if (fmiType == 1) {
     FMI2ModelExchange* FMI2ME = (FMI2ModelExchange*)in_fmi2;
-    if (FMI2ME->FMISolvingMode == fmi2_instantiated_mode || FMI2ME->FMISolvingMode == fmi2_initialization_mode || FMI2ME->FMISolvingMode == fmi2_event_mode) {
+    if (FMI2ME->FMISolvingMode == fmi2_instantiated_mode || FMI2ME->FMISolvingMode == fmi2_initialization_mode || FMI2ME->FMISolvingMode == fmi2_event_mode || FMI2ME->FMISolvingMode == fmi2_continuousTime_mode) {
```
[the remark this answers is not in the document]
Comment (AnHeuermann) [reply OMCompiler/SimulationRuntime/c/fmi/FMI2Common.c:159]: > So why not allow it in OpenModelica too? Is there any reason not to do so, besides a formal violation of the standard?

Well, I guess not. But there is a reason FMI and Modelica are standardized in the first place. If everyone would follow a (non existing) perfect standard each model would always work in every tool. Of course this is unrealistic, but following the standard is maybe the best option we have.

@casella Do we have an official approach on this?

Someone could open a change request for FMI 2.0.5 to also allow it.

[the remark this answers is not in the document]
Comment (AnHeuermann) [reply OMCompiler/SimulationRuntime/c/fmi/FMI2Common.c:159]: The only thing I can see for FMUs linked like

grafik

to work is to create events for each change of an integer/boolean. Even if they are wrapped with `noEvent`, which is stupid.

Allowing the non-standard function call would be way more easy. But then we need to do it for OMSimulator as well...

**Comment by casella** [comment]
If it was dropped in 3.0 there was probably a good reason for it. But I would definitely recommend to propose a fix for 2.0.5. Please go ahead with that.

**Comment by nik5346** [comment]
@AnHeuermann 
I can't reopen the branch. Here is the branch to merge:
https://github.com/nik5346/OpenModelica/tree/fmi2SetXXX_cont_time

**Pull request closed without merging**